tephra-core/src/main/java/org/apache/tephra/util/TxUtils.java

```diff
@@ -75,12 +75,9 @@ public static long getOldestVisibleTimestamp(Map<byte[], Long> ttlByFamily, Tran
    * @return The oldest timestamp that will be visible for the given transaction and TTL configuration
    */
   public static long getOldestVisibleTimestamp(Map<byte[], Long> ttlByFamily, Transaction tx, boolean readNonTxnData) {
-    if (readNonTxnData) {
-      long maxTTL = getMaxTTL(ttlByFamily);
-      return maxTTL < Long.MAX_VALUE ? System.currentTimeMillis() - maxTTL : 0;
-    }
-
-    return getOldestVisibleTimestamp(ttlByFamily, tx);
+    long ttlFactor = readNonTxnData ? 1 : TxConstants.MAX_TX_PER_MS;
+    long maxTTL = getMaxTTL(ttlByFamily);
```

Comment:
the `maxTTL` should be capped by `Math.min(maxTTL, Long.MAX_VALUE / ttlFactor)`, right?

Reply:
I think it should instead get capped by a lower value. In any case, I pushed a change that modifies this code. Please take another look.